src/inline-loader.ts

```diff
@@ -73,31 +74,31 @@ module.exports = async function loader(
       source: string;
       styleId: string;
     };
-    this.resourcePath = source;
+
     const styles = await cache.get(source);
 
-    let css = styles?.get(styleId)?.value;
+    let style = styles?.get(styleId);
 
-    if (!css) {
+    if (!style) {
       await loadModule(source);
-      css = inMemoryStyleCache.get(source)?.get(styleId)?.value;
+      style = inMemoryStyleCache.get(source)?.get(styleId);
     }
 
-    if (!css) {
+    if (!style) {
       return cb(
         new Error(
           `Could not resolve style ${loaderOpts.styleId} in file ${loaderOpts.source}`,
         ),
       );
     }
-
-    return cb(null, css);
+    this._module.matchResource = style.absoluteFilePath;
```

Comment:
this "fixes" the issue for css-loader, but seems really iffy. Maybe it's fine and also the simplest solution tho

Reply:
hm, I don't think we need do it, it potential break other cases

Reply:
another way to solve this might be to have css-loader not look at matchResource and instead read something out of `meta` or a query param in the request?

Reply:
Yes, it is possible to improve, but why?

Reply:
just don't touch this and all should work

Reply:
if it worked i'd happily leave it alone. The problem is it doesn't work when you use a real webpack config with style-loader or mini-extract-css-plugin

here's a simplified reproduction: https://github.com/jquense/match-resource-repro

Reply:
(i'm really not trying to make a lot of work for you, if you think there is a reasonable path here i'd be happy to code it up, otherwise i can do my hacks)

Reply:
Yep, we are losing `matchSource` here, because `request` was rewritten, let's discuss about it here https://github.com/webpack/webpack/issues/11074